src/backend/web-gl/function-node.js

```diff
@@ -177,14 +181,14 @@ module.exports = class WebGLFunctionNode extends FunctionNodeBase {
 	 */
 	astReturnStatement(ast, retArr) {
 		if (this.isRootKernel) {
-			retArr.push('kernelResult = ');
+			retArr.push('kernelResult = float(');
```

Comment:
Couldn't we move the casting into the js side, rather than it being casted in glsl?

Reply:
yes, it seems that this is unnecessary - i suppose i wanted to catch the possibility of `return this.thread.z` which is a pretty unlikely and pointless scenario thinking about it.  All test cases pass without it!

if it comes a problem in future I guess we need a 'force-float' state and add a cast if necessary